### ISSUE TITLE
value/reward transform issue

### ISSUE THREAD

**Issue by xuxiyang1993** [werner-duvaud/muzero-general#6]
I think the way you transform value/reward is a little mismatch with the original paper at this line (https://github.com/werner-duvaud/muzero-general/blob/fe791e8651645ea05f5b582157b4892588ee56ca/trainer.py#L153)

From the referenced paper (https://arxiv.org/abs/1805.11593), the transformation function should be 
<img width="346" alt="transform" src="https://user-images.githubusercontent.com/29387830/74098329-93229880-4adc-11ea-9705-668d8e164a9a.png">

So instead of

`x = torch.sign(x) * (torch.sqrt(torch.abs(x) + 1) - 1 + 0.001 * x)`

the correct formula should be

`x = torch.sign(x) * (torch.sqrt(torch.abs(x) + 1) - 1) + .001 * x`

**Comment by xuxiyang1993** [comment]
Also at this line (https://github.com/werner-duvaud/muzero-general/blob/master/trainer.py#L157), if x has an integer value (which may happen if value/reward is out of `support_size` range and got clamped), `floor` and `ceil` will have same value, then `prob=0`, then the final logits will be set twich

**Comment by werner-duvaud** [comment]
You are right, thank you.

Indeed, we had implemented the formula which is written in MuZero paper (appendix Network Architecture), but this formula is different from the one in the paper [Observe and Look Further: Achieving Consistent Performance on Atari](https://arxiv.org/abs/1805.11593).

It was fixed in the commit b1bd9ab.

**Comment by xuxiyang1993** [comment]
I see, I feel the formula in MuZero paper is a typo when they cite it. After reading the original paper I think transformation and inverse transformation formulas from Lemma A1 & A2 are correct.

By the way thanks for your implementation of the MuZero algorithm, it really helps me a lot!
